python/foxglove-sdk/build.rs

```diff
@@ -30,54 +29,47 @@ fn compile_protos() {
         ));
     }
 
-    /*
-    // Compile protos
-    let mut command = Command::new("protoc");
+    // Write __init__.py
+    let init_path = out_dir.join("__init__.py");
+    fs::write(init_path, init_imports.join("\n") + "\n").unwrap();
+}
+
+fn rustfmt(file: &Path) {
+    let mut command = Command::new("cargo");
     command
-        .arg(format!("--proto_path={}", proto_path.display()))
-        .arg(format!("--python_out={}", out_dir.display()))
-        .arg(format!("--pyi_out={}", out_dir.display()));
+        .arg("fmt")
+        .arg("--")
+        .arg(format!("{}", file.display()));
 
-    // Add all proto files to the command
-    for proto_file in &proto_files {
-        println!("cargo:rerun-if-changed={}", proto_file.display());
-        command.arg(proto_file);
-    }
+    let status = command.status().expect("cargo fmt should have run");
 
-    let status = command
-        .status()
-        .map_err(|e| match e.kind() {
-            ErrorKind::NotFound => {
-                panic!("The command `protoc` was not found. Probably you need to install it.")
-            }
-            _ => panic!("protoc failed: {}", e),
-        })
-        .unwrap();
     if !status.success() {
-        panic!("protoc failed!");
+        panic!("cargo fmt failed");
     }
+}
 
-    // Fix imports in generated .py files
-    for proto_file in proto_files {
-        // python name will look like ./foxglove/ProtoName_pb2.py
-        let py_file = out_dir.join("foxglove").join(format!(
-            "{}_pb2.py",
-            proto_file.file_stem().unwrap().to_str().unwrap()
-        ));
+fn import_schemas() {
+    let src_dir = Path::new("../../schemas/pyclass");
+    let rust_dir = Path::new("src");
+    let python_dir = Path::new("python/foxglove");
 
-        let mut file = std::fs::File::open(&py_file).unwrap();
-        let mut contents = String::new();
-        file.read_to_string(&mut contents).unwrap();
-        let new_contents = contents.replace("from foxglove import ", "from . import ");
-        std::fs::write(py_file, new_contents).unwrap();
-    }
-    */
+    fs::create_dir_all(rust_dir).unwrap();
 
-    // Write __init__.py
-    let init_path = out_dir.join("__init__.py");
-    std::fs::write(init_path, init_imports.join("\n") + "\n").unwrap();
+    // pyo3 schema definitions
+    let src = Path::join(src_dir, "schemas.rs");
+    let dest = Path::join(rust_dir, "schemas.rs");
+    println!("cargo:rerun-if-changed={}", src.display());
+    fs::copy(&src, &dest).expect("schemas.rs should be copied by build script");
+    rustfmt(&dest);
+
+    // python stub interface
+    let src = Path::join(src_dir, "schemas.pyi");
+    let dest = Path::join(python_dir, "_foxglove_py/schemas.pyi");
+    println!("cargo:rerun-if-changed={}", src.display());
+    fs::copy(&src, &dest).expect("schemas.pyi should be copied by build script");
```

Comment:
seems simpler if we can move all this (including `rustfmt`) into the generate script